x-pack/plugins/ml/public/application/services/get_shared_ml_services.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { type HttpStart } from '@kbn/core-http-browser';
+import { type UsageCollectionSetup } from '@kbn/usage-collection-plugin/public';
+import { ElasticModels } from './elastic_models_service';
+import { HttpService } from './http_service';
+import { mlApiServicesProvider } from './ml_api_service';
+import { mlUsageCollectionProvider } from './usage_collection';
+import { MlCapabilitiesService } from '../capabilities/check_capabilities';
+import { MlLicense } from '../../../common/license';
+
+export type MlSharedServices = ReturnType<typeof getMlSharedServices>;
+
+/**
+ * Provides global services available across the entire ML app.
+ */
+export function getMlSharedServices(
+  httpStart: HttpStart,
+  isServerless: boolean,
+  usageCollection?: UsageCollectionSetup
+) {
+  const httpService = new HttpService(httpStart);
+  const mlApiServices = mlApiServicesProvider(httpService);
+
+  return {
+    httpService,
+    mlApiServices,
+    mlUsageCollection: mlUsageCollectionProvider(usageCollection),
+    mlCapabilities: new MlCapabilitiesService(mlApiServices),
+    mlLicense: new MlLicense(),
+    elasticModels: new ElasticModels(mlApiServices.trainedModels),
+    isServerless,
```

Comment:
For now should we just return the items which are being used? which at the moment is only `elasticModels`.
We can add items back in when there is a need for them.

We definitely shouldn't be sharing an `isServerless` flag.

Reply:
I'm not sharing it, only `elasticModels ` is returned in `start`

Reply:
so shall we remove all other items from this returned object?

Reply:
updated in https://github.com/elastic/kibana/pull/167000/commits/04134c69f0210e5fbf4c0c689f247ddb3da0dc27